src/main/java/com/yanna/stepanova/service/impl/BookServiceImpl.java

```diff
@@ -1,28 +1,78 @@
 package com.yanna.stepanova.service.impl;
 
+import com.yanna.stepanova.dto.BookDto;
+import com.yanna.stepanova.dto.CreateBookRequestDto;
+import com.yanna.stepanova.exception.EntityNotFoundException;
+import com.yanna.stepanova.mapper.BookMapper;
 import com.yanna.stepanova.model.Book;
 import com.yanna.stepanova.repository.BookRepository;
 import com.yanna.stepanova.service.BookService;
 import java.util.List;
-import org.springframework.beans.factory.annotation.Autowired;
+import java.util.Random;
+import lombok.RequiredArgsConstructor;
 import org.springframework.stereotype.Service;
 
+@RequiredArgsConstructor
 @Service
 public class BookServiceImpl implements BookService {
     private final BookRepository bookRepo;
+    private final BookMapper bookMapper;
+    private final Random random = new Random();
```

Comment:
You are using a Spring. So please, make Random as Bean, do not create an instance by yourself